pulsar-client-cpp/tests/BasicEndToEndTest.cc

```diff
@@ -815,6 +826,6 @@ TEST(BasicEndToEndTest, testMessageListenerPause)
 
     // 3 seconds
     usleep(3 * 1000 * 1000);
-
+    Lock lock(mutex_);
```

Comment:
There are two threads using the variable - hence W/O the lock the behavior is undefined.